src/components/wizard/WizardStep.js

```diff
@@ -0,0 +1,159 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {StyleSheet} from 'react-native';
+import View from '../view';
+import Text from '../text';
+import Image from '../image';
+import TouchableOpacity from '../touchableOpacity';
+import {PureBaseComponent} from '../../commons';
+import Colors from '../../style/colors';
+import BorderRadiuses from '../../style/borderRadiuses';
+import Spacings from '../../style/spacings';
+import {States, StatesConfig} from './WizardStates';
+
+/**
+ * @description: WizardStep Component: a wizard presents a series of steps in  prescribed order
+ * that the user needs to complete in order to accomplish a goal (e.g. purchase a product).
+ *
+ * @example: https://github.com/wix-private/wix-react-native-ui-lib/blob/master/example/screens/components/WizardScreen.js
+ * @guidelines: https://zpl.io/aXmAkdg
+ * @notes: Use Wizard with nested Wizard.Step(s) to achieve the desired result.
+ */
+export default class WizardStep extends PureBaseComponent {
+  static displayName = 'Wizard.Step';
+
+  static propTypes = {
+    /**
+     * The label of the item
+     */
+    label: PropTypes.string,
+    /**
+     * Additional styles for the label
+     */
+    labelStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
+    /**
+     * Additional styles for the connector
+     */
+    connectorStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
+    /**
+     * Color of the step index (or of the icon, when provided)
+     */
+    color: PropTypes.string,
+    /**
+     * Color of the circle
+     */
+    circleColor: PropTypes.string,
+    /**
+     * Icon to replace the (default) index
+     */
+    icon: PropTypes.oneOfType([PropTypes.object, PropTypes.number]),
+    /**
+     * Additional styles for the index's label (when icon is not provided)
+     */
+    indexLabelStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
+    /**
+     * Whether the step should be enabled
+     */
+    enabled: PropTypes.bool
+  };
+
+  getProp(activeConfig, config, propName) {
+    const props = this.getThemeProps();
```

Comment:
why extracting the props like that?
also `index` and `activeIndex` can be taken directly from `this.props`

Reply:
ok I see why u did that. 

anyway, you can shorten the way you get the config values. 
the way I see it there are 3 sets of config. 
the state config, the active config and the props (which should are identical to the config structure)

you can create some sort of merged config out of those three and just take the final values from the merge config. 
it should turn this whole code two 2-3 lines of code (:

Reply:
Done.